pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -307,22 +314,49 @@ func createHTTPMatchRequest(prefix, targetHost, internalHost string, additionalH
 
 		if additionalHosts != nil && len(*additionalHosts) != 0 {
 			for _, host := range *additionalHosts {
-				matchRequests = append(matchRequests,
-					&istiov1beta1.HTTPMatchRequest{
-						Uri: uri,
-						Authority: &istiov1beta1.StringMatch{
-							MatchType: &istiov1beta1.StringMatch_Regex{
-								Regex: constants.HostRegExp(host),
-							},
+				matchRequest := &istiov1beta1.HTTPMatchRequest{
+					Uri: uri,
+					Authority: &istiov1beta1.StringMatch{
+						MatchType: &istiov1beta1.StringMatch_Regex{
+							Regex: constants.HostRegExp(host),
 						},
-						Gateways: []string{config.IngressGateway},
-					})
+					},
+					Gateways: []string{config.IngressGateway},
+				}
+				if !containsHTTPMatchRequest(matchRequest, matchRequests) {
+					matchRequests = append(matchRequests, matchRequest)
+				}
 			}
 		}
 	}
 	return matchRequests
 }
 
+func containsHTTPMatchRequest(matchRequest *istiov1beta1.HTTPMatchRequest, matchRequests []*istiov1beta1.HTTPMatchRequest) bool {
+	for _, matchRequestEle := range matchRequests {
+		// If authority, gateways and uri are all equal, two HTTPMatchRequests will be equal.
+		if stringMatchEqual(matchRequest.Authority, matchRequestEle.Authority) && gatewaysEqual(matchRequest, matchRequestEle) &&
+			stringMatchEqual(matchRequest.Uri, matchRequestEle.Uri) {
+			return true
+		}
```

Comment:
So we can't perform deepEqual directly on `HTTPMatchRequest` ?

Reply:
No, I tried. It seems I have to break them and check the field one  by one with this approach.